pkg/agent/utils.go

```diff
@@ -364,7 +382,7 @@ func GetOrderedKubeletConfigFlagString(k *datamodel.KubernetesConfig, cs *datamo
 	return buf.String()
 }
 
-// IsDynamicKubeletEnabled get if dynamic kubelet is supported in AKS
+// IsDynamicKubeletEnabled get if dynamic kubelet is supported in AKS and toggle is on
 func IsDynamicKubeletEnabled(cs *datamodel.ContainerService, dynamicKubeletToggleEnabled bool) bool {
```

Comment:
I really don't like this dynamicKubelet concept confusion, let's update it in next PR if not this

Reply:
Sure I'll do in a later PR. It'll change quite a few files.